src/testing/core/mocking/__tests__/mockLink.ts

```diff
@@ -0,0 +1,32 @@
+import gql from "graphql-tag";
+import { MockLink } from "../mockLink";
+import { createOperation } from "../../../../link/utils";
+
+describe("mockLink", () => {
+  it("should not require a result or error when delay equals Infinity", async () => {
+    const query = gql`
+      query A {
+        a
+      }
+    `;
+    const mockLink = new MockLink([
+      {
+        request: {
+          query,
+        },
+        delay: Infinity,
+      },
+    ]);
+
+    const observable = mockLink.request(createOperation(null, { query }));
+
+    expect(observable).not.toBeNull();
+    const subscription = observable!.subscribe(
+      () => fail("onNext was called"),
+      () => fail("onError was called"),
+      () => fail("onComplete was called")
+    );
+    await new Promise((resolve) => setTimeout(resolve, 10));
```

Comment:
We have a [`wait` helper](https://github.com/apollographql/apollo-client/blob/7555b2e2139049b3ce21198cb896ccc105dfadb8/src/testing/core/wait.ts#L1-L3) that you can use here for this delay which you can get from the [`testing` entry point](https://github.com/apollographql/apollo-client/blob/main/src/testing/index.ts). Would you mind switching to this?

```suggestion
    await wait(10);
```